sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -362,12 +362,46 @@ public virtual Response Delete(string partitionKey, string rowKey, string eTag =
             Argument.AssertNotNull(rowKey, nameof(rowKey));
 
             return _tableOperations.DeleteEntity(_table,
-                                          partitionKey as string,
-                                          rowKey as string,
+                                          partitionKey,
+                                          rowKey,
                                           queryOptions: new QueryOptions() { Format = _format, ETag = eTag },
                                           cancellationToken: cancellationToken);
         }
 
+        /// <summary> Retrieves details about any stored access policies specified on the table that may be used wit Shared Access Signatures. </summary>
```

Comment:
typo: `wit` -> `with`

And are these ACLs only used with shared access signatures?  We could use them for more than that in other Storage libraries.